src/passes/LegalizeJSInterface.cpp

```diff
@@ -241,38 +242,53 @@ struct LegalizeJSInterface : public Pass {
     return func->name;
   }
 
-  void ensureTempRet0(Module* module) {
+
+  void addTempRet0Helpers(Module* module) {
+    bool getterExists = module->getExportOrNull(GET_TEMP_RET_0) != nullptr;
+    bool setterExists = module->getExportOrNull(SET_TEMP_RET_0) != nullptr;
+
     if (!module->getGlobalOrNull(TEMP_RET_0)) {
```

Comment:
Is the idea that if the `tempRet0` global appears, we must also already have the getter and setter for it, to avoid mixing a getter for one mechanism with another? I think we should also check that if the global does not appear, that neither the getter nor the setter do either.

Reply:
Yes, thats the idea.    I've added more checks and comments that hopefully make it more clear.